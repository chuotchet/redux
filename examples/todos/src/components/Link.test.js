import React from 'react';
import { shallow } from 'enzyme';

import Link from './Link';

const setup = (setupProps = {}) => {
    const defaultProps = {
        active: false,
        onClick: jest.fn(),
        children: 'Test'
    }
    const props = {...defaultProps,...setupProps};
    const wrapper = shallow(
        <Link onClick={props.onClick} active={props.active} children={props.children} />);

    return {
        props,
        wrapper
    };
};

describe('Link', () => {
    it('should match snapshot', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    })
    it('should handle click button', () => {
        const { props, wrapper } = setup();

        const link = wrapper.shallow().find('button');
        link.simulate('click');

        expect(props.onClick).toBeCalled();
    })
})