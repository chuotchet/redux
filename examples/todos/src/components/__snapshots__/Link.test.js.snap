// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
Object {
  "props": Object {
    "active": false,
    "children": "Test",
    "onClick": [MockFunction],
  },
  "wrapper": ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Link
      active={false}
      onClick={[MockFunction]}
    >
      Test
    </Link>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
        "disabled": false,
        "onClick": [MockFunction],
        "style": Object {
          "marginLeft": "4px",
        },
      },
      "ref": null,
      "rendered": "Test",
      "type": "button",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
          "disabled": false,
          "onClick": [MockFunction],
          "style": Object {
            "marginLeft": "4px",
          },
        },
        "ref": null,
        "rendered": "Test",
        "type": "button",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
}
`;
